neural_style.py

```diff
@@ -200,7 +201,19 @@ def imread(path):
 
 def imsave(path, img):
     img = np.clip(img, 0, 255).astype(np.uint8)
-    Image.fromarray(img).save(path, quality=95)
+    im_obj = Image.fromarray(img)
+    try:
+        im_obj.save(path, quality=95)
+    except KeyError:
```

Comment:
Falling back to a default path may result in data loss -- e.g. you already have a file called `foo.png`, and you try to save a new thing as `foo.jpg`, but you mistype it as `foo.jgp`, and then this overwrites your `foo.png`.

I think a better solution would be to validate the image extension against a whitelist, in the same place where you check the `is_path_writable`.